trade/trader_stakecube.js

```diff
@@ -0,0 +1,664 @@
+
+      const marketInfo = this.marketInfo(pair);
+
+      let message;
+
+      if (!marketInfo) {
+        message = `Unable to place an order on ${exchangeName} exchange. I don't have info about market ${pair}.`;
+        log.warn(message);
+        return {
+          message,
+        };
+      }
+
+      // for Limit orders, calculate coin1Amount if only coin2Amount is provided
+      if (!coin1Amount && coin2Amount && price) {
+        coin1Amount = coin2Amount / price;
+      }
+
+      // for Limit orders, calculate coin2Amount if only coin1Amount is provided
+      let coin2AmountCalculated;
+      if (!coin2Amount && coin1Amount && price) {
+        coin2AmountCalculated = coin1Amount * price;
+      }
+
+      if (coin1Amount) {
+        coin1Amount = +(+coin1Amount).toFixed(marketInfo.coin1Decimals);
+      }
+      if (coin2Amount) {
+        coin2Amount = +(+coin2Amount).toFixed(marketInfo.coin2Decimals);
+      }
+      if (price) {
+        price = +(+price).toFixed(marketInfo.coin2Decimals);
+      }
+
+      if (coin1Amount && coin1Amount < marketInfo.coin1MinAmount) { // coin1Amount may be null
+        message = `Unable to place an order on ${exchangeName} exchange. Order amount ${coin1Amount} ${marketInfo.coin1} is less minimum ${marketInfo.coin1MinAmount} ${marketInfo.coin1} on ${marketInfo.pairReadable} pair.`;
+        log.warn(message);
+        return {
+          message,
+        };
+      }
+
+      if (coin2Amount && coin2Amount < marketInfo.coin2MinAmount) { // coin2Amount may be null, and skip coin2AmountCalculated checking, it's for market order only
+        message = `Unable to place an order on ${exchangeName} exchange. Order volume ${coin2Amount} ${marketInfo.coin2} is less minimum ${marketInfo.coin2MinAmount} ${marketInfo.coin2} on ${marketInfo.pairReadable} pair.`;
+        log.warn(message);
+        return {
+          message,
+        };
+      }
+
+      const order = {};
+      let output;
+
+      if (limit) { // Limit order
+        const pairName = formatPairName(pair);
+        output = `${side} ${coin1Amount} ${pairName.coin1} at ${price} ${pairName.coin2}.`;
+
+        const order = {};
+        let orderId;
+        let errorMessage;
+        let response;
+
+        try {
+          response = await stakeCubeApiClient.addOrder(marketInfo.pairPlain, coin1Amount, price, side);
+
+          orderId = response?.orderId;
+          errorMessage = response?.errorMessage;
+        } catch (err) {
+          message = `API request addOrder(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${err}.`;
+          log.warn(message);
+          order.orderId = false;
+          order.message = message;
+
+          return order;
+        }
+
+        if (orderId) {
+          message = `Order placed to ${output} Order Id: ${orderId}.`;
+          log.info(message);
+          order.orderId = orderId;
+          order.message = message;
+        } else if (!orderId && !errorMessage) {
```

Comment:
Looks bad.
Do you mean it's possible to place an order, but receive no `orderId`?

Reply:
I've tested placing orders. When order is fullfilled immediately after placing, StakeCube returns no orderId

Reply:
I've sent a request to fix it to StakeCube dev.